shopinvader_membership/services/membership.py

```diff
@@ -41,33 +42,51 @@ def search(self, **params):
         """
         return self._paginate_search(**params)
 
-    def subscribe(self, _id):
+    @restapi.method(
+        [(["/<int:_id>/subscribe"], "GET")],
+        input_param=restapi.CerberusValidator({}),
+        output_param=restapi.CerberusValidator("_validator_return_subscribe"),
+    )
+    def get_subscribe(self, _id):
         """
         Subscribe to a membership product with logged user
         :param _id: id of product.product
         :return: dict with invoice_id
         """
+        product_id = _id
         if not self._is_logged_in():
             raise UserError(_("A user should be logged"))
         membership_product = self.env["product.product"].search(
-            [("id", "=", _id), ("membership", "=", True)]
+            [("id", "=", product_id), ("membership", "=", True)], limit=1
         )
         if not membership_product:
-            raise UserError(_("No membership product found with id %s") % _id)
+            raise UserError(_("No membership product found with id %s") % product_id)
         wizard = self.env["membership.invoice"].create(
-            {"product_id": _id, "member_price": membership_product.list_price}
+            {"product_id": product_id, "member_price": membership_product.list_price}
         )
         invoices_views_dict = wizard.with_context(
             active_ids=self.partner.ids
         ).membership_invoice()
         return {"invoice_id": invoices_views_dict.get("domain")[0][2][0]}
 
+    @restapi.method(
+        [(["/subscribe"], "POST")],
+        input_param=restapi.CerberusValidator("_validator_subscribe"),
+        output_param=restapi.CerberusValidator("_validator_return_subscribe"),
+    )
+    def post_subscribe(self, **params):
+        """
+        Subscribe to a membership product with logged user
+        :return: dict with invoice_id
+        """
+        return self.get_subscribe(_id=params.get("membership_product_id"))
+
     def _validator_subscribe(self):
         """
         Validator for the subscribe
         :return: dict
         """
-        return {"membership_product_id": {"type": "integer"}}
+        return {"membership_product_id": {"type": "integer", "required": False}}
```

Comment:
```suggestion
        return {"membership_product_id": {"type": "integer", "required": True}}
```

I suppose it does make sense to subscribe without a membership product